packages/xchain-polkadot/src/client.ts

```diff
@@ -15,53 +15,36 @@ import {
 import { Asset, assetAmount, assetToString, assetToBase, baseAmount } from '@xchainjs/xchain-util/lib'
 import * as xchainCrypto from '@xchainjs/xchain-crypto'
 
-import Keyring from '@polkadot/keyring'
+import { ApiPromise, WsProvider, Keyring } from '@polkadot/api'
 import { KeyringPair } from '@polkadot/keyring/types'
-import { RuntimeDispatchInfo } from '@polkadot/types/interfaces'
-import HttpProvider from '@polkadot/rpc-provider/http'
-import * as txWrapper from '@substrate/txwrapper'
-import { EXTRINSIC_VERSION } from '@substrate/txwrapper/lib/util/constants'
 
 import { SubscanResponse, Account, AssetDOT, TransfersResult, Extrinsic, Transfer } from './types'
 import { isSuccess } from './util'
 
-const {
-  POLKADOT_SS58_FORMAT,
-  WESTEND_SS58_FORMAT,
-  methods,
-  createSigningPayload,
-  createSignedTx,
-  getRegistry,
-} = txWrapper
-
 const DECIMAL = 10
 
 /**
  * Interface for custom Polkadot client
  */
 export interface PolkadotClient {
   getSS58Format(): number
-  getChainName(): 'Polkadot' | 'Westend'
-  getSpecName(): 'polkadot' | 'westend'
-
+  getWsEndpoint(): string
   estimateFees(params: TxParams): Promise<Fees>
 }
 
 class Client implements PolkadotClient, XChainClient {
   private network: Network
   private phrase = ''
   private address: Address = ''
-  private provider: HttpProvider
+  private api: ApiPromise | null = null
 
   constructor({ network = 'testnet', phrase }: XChainClientParams) {
     this.network = network
 
-    this.provider = new HttpProvider(this.getRPCEndpoint())
-
     if (phrase) this.setPhrase(phrase)
   }
 
-  purgeClient(): void {
+  purgeClient = (): void => {
```

Comment:
Don't we need to `api.disconnect()` here, if `api` is not null? As you suggested before ...